src/gui/gtk.c

```diff
@@ -4328,8 +4376,10 @@ void dt_gui_cursor_clear_busy()
 
 void dt_gui_process_events()
 {
-  // process all pending Gtk/GDK events
-  while(g_main_context_iteration(NULL, FALSE))
+  // process pending Gtk/GDK events; we need to limit the total calls because once the LUA
+  // interpreeter starts the script installer we would end up in an infinite loop
+  unsigned max_iter = 200;
+  while(g_main_context_iteration(NULL, FALSE) && --max_iter > 0)
```

Comment:
@ralfbrown Is there a particular reason to use `g_main_context_iteration` rather than the more standard
```
while(gtk_events_pending()) gtk_main_iteration();
```
? Are there any non-event context iterations that _must_ be handled?

I haven't done too much in-depth testing of corner cases, but ba23d75c33ff1406c7249f9ea3ee4a77df708194 seems to handle lua initialisation just fine and I haven't noticed any other problematic behavior.

Reply:
I'm new to Gtk and that's what I found....
Will take a look over the weekend.